### DISCUSSION TITLE
Error 403 Request had insufficient authentication scopes.

### DISCUSSION THREAD

**Discussion by ntseles** [spatie/laravel-google-calendar#259] [General]
I have spent the last 2 days trying to use this in my application. I have read over everything in the md file and searched for hours. This is the response I'm getting from Google.

 "error": {
        "code": 403,
        "message": "Request had insufficient authentication scopes.",
        "errors": [
            {
                "message": "Insufficient Permission",
                "domain": "global",
                "reason": "insufficientPermissions"
            }
        ],
        "status": "PERMISSION_DENIED",
        "details": [
            {
                "@type": "type.googleapis.com/google.rpc.ErrorInfo",
                "reason": "ACCESS_TOKEN_SCOPE_INSUFFICIENT",
                "domain": "googleapis.com",
                "metadata": {
                    "service": "calendar-json.googleapis.com",
                    "method": "calendar.v3.Events.Insert"
                }

**Comment by aawMnt** [comment]
i have also same error can any one find the solution

403 Forbidden
{
  "code" : 403,
  "errors" : [ {
    "domain" : "global",
    "message" : "Insufficient Permission",
    "reason" : "insufficientPermissions"
  } ],
  "message" : "Request had insufficient authentication scopes.",
  "status" : "PERMISSION_DENIED",
  "details" : [ {
    "@type" : "type.googleapis.com/google.rpc.ErrorInfo",
    "reason" : "ACCESS_TOKEN_SCOPE_INSUFFICIENT",
    "domain" : "googleapis.com",
    "metadata" : {
      "service" : "youtube.googleapis.com",
      "method" : "youtube.api.v3.V3DataPlaylistItemService.List"
    }
  } ]
}

this google API working properly I am not changing anything but suddenly this error occurs

**Reply by EmilioPuljiz** [reply]
What you have to do is in the developer console you must give permission to the Google Calendar API so that external users can edit the events, and there you have to revoke the service account user in the calendar and re-enable it with the new one. permission, be careful if or if you have to remove it and add it again, I wanted to edit the permissions and it didn't work either
